### DISCUSSION TITLE
Adding elements to the netlist

### DISCUSSION THREAD

**Discussion by maxdd** [nunobrum/PyLTSpice#145] [Q&A]
How am i suppose to add Resistors, Capactitors or a voltage source the proper way?
Do i have to do

```
class Component():
    def __init__(self, reference, ports, value):
        self.reference = reference
        self.ports = ports
        self.attributes = {'model' : value}
    
netlist.add_component(Component(reference='V10', ports=['n1', 'n2'] , value='0v') )
```

I'm asking this because add_component feels like is intended for external components, in fact i have to be a bit hacky with that '{'model' : value}' otherwise i would get a "no_model" keyword in the netlist

**Comment by nunobrum** [comment]
PyLTSpice was not thought with the purpose of creating netlists using python. 
My own use case is normally to manipulate components that already exist on the circuit. 
If you are familiar with SPICE, it is way simpler to write the netlist using SPICE, or using a schematic editor.

If you really need to use PyLTSpice, then just do this:

`netlist.add_instruction("V10 n1 n2 0V")`

This is why I never bothered with this feature. PyLTSpice and spicelib are just tools to automate simulations once the netlist is created.

**Reply by maxdd** [reply]
Got it
